include/picongpu/fields/MaxwellSolver/ArbitraryOrderFDTD/Derivative.hpp

```diff
@@ -0,0 +1,234 @@
+     *                     the derivative from finite differences.
+     *                     Order of derivative approximation is
+     *                     2 * T_neighbors
+     *
+     * @tparam T_direction Direction to take derivative in, 0 = x, 1 = y, 2 = z
+     */
+    template<
+        uint32_t T_lowerNeighbors,
+        uint32_t T_upperNeighbors,
+        uint32_t T_neighbors,
+        uint32_t T_direction
+     >
+    struct GeneralAofdtdDerivative
+    {
+        //! Lower margin
+        using LowerMargin = typename pmacc::math::CT::mul<
+            typename pmacc::math::CT::make_Int<
+                simDim,
+                T_lowerNeighbors
+            >::type,
+            typename pmacc::math::CT::make_BasisVector<
+                simDim,
+                T_direction,
+                int
+            >::type
+        >::type;
+
+        //! Upper margin
+        using UpperMargin = typename pmacc::math::CT::mul<
+            typename pmacc::math::CT::make_Int<
+                simDim,
+                T_upperNeighbors
+            >::type,
+            typename pmacc::math::CT::make_BasisVector<
+                simDim,
+                T_direction,
+                int
+            >::type
+        >::type;
+
+        /** Return derivative value at the given point
+         *
+         * @tparam T_DataBox data box type with field data
+         * @param data position in the data box to compute derivative at
+         */
+        template< typename T_DataBox >
+        HDINLINE typename T_DataBox::ValueType operator()(
+            T_DataBox const & data ) const
+        {
+            // Define shorthand type to access DataBox
+            using IndexType = pmacc::DataSpace< simDim >;
+
+            // Define indice vectors for data access
+            auto lowerIndex = IndexType{}; // Vector initialized with zeros
+            auto upperIndex = IndexType{};
+
+            // lowerIndex: 0 if ( Forward ) else -1
+            lowerIndex[ T_direction ] = static_cast<int32_t>(T_upperNeighbors) - static_cast<int32_t>(T_neighbors);
+            // upperIndex: 1 if ( Forward ) else 0
+            upperIndex[ T_direction ] = static_cast<int32_t>(T_neighbors) - static_cast<int32_t>(T_lowerNeighbors);
+
+            AOFDTDWeights< T_neighbors > const weights{};
+
+            // shortest distance finite difference as initial value
+            auto finiteDifference = weights[0] *
+                ( data( upperIndex ) - data( lowerIndex ) );
+
+            // Compute next finite differences according to order
+            for ( uint32_t l = 1; l < T_neighbors; ++l )
+            {
+                lowerIndex[ T_direction ] -= 1;
+                upperIndex[ T_direction ] += 1;
+
+                finiteDifference += weights[l] *
+                    ( data( upperIndex ) - data( lowerIndex ) );
+            }
+
+            return finiteDifference / cellSize[ T_direction ];
+        }
+    };
+} // namespace detail
+
+
+    /**@{*/
```

Comment:
Group comment, suggested by psychocoderHPC since the documentation of the following two definitions is almost equal. Allows two have one doxygen comment string for multiple following definitions. See (https://github.com/ComputationalRadiationPhysics/picongpu/pull/3338#discussion_r496463295)

Reply:
This is for a group